state/remoteentities.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	jujutxn "github.com/juju/txn"
+	"github.com/juju/utils"
+	"gopkg.in/juju/names.v2"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+// remoteEntityDoc represents the internal state of a remote entity in
+// MongoDB. Remote entities may be exported local entities, or imported
+// remote entities.
+type remoteEntityDoc struct {
+	DocID string `bson:"_id"`
+
+	SourceModelUUID string `bson:"source-model-uuid"`
+	EntityTag       string `bson:"entity"`
+	Token           string `bson:"token"`
+}
+
+type tokenDoc struct {
+	Token     string `bson:"_id"`
+	ModelUUID string `bson:"model-uuid"`
+}
+
+// RemoteEntities wraps State to provide access
+// to the remote entities collection.
+type RemoteEntities struct {
+	st *State
+}
+
+// RemoteEntities returns a wrapped state instance providing
+// access to the remote entities collection.
+func (st *State) RemoteEntities() *RemoteEntities {
+	return &RemoteEntities{st}
+}
+
+// ExportLocalEntity adds an entity to the remote entities collection,
+// returning an opaque token that uniquely identifies the entity within
+// the model.
+//
```

Comment:
Disregard I just tried a linter to see and it likes it, I was curious about the rule for Docs and how do those behave with newlines.